airflow/jobs/scheduler_job.py

```diff
@@ -1295,13 +1293,19 @@ def _execute(self) -> None:
                 )
                 models.DAG.deactivate_stale_dags(execute_start_time)
 
-            self.executor.end()
-
             settings.Session.remove()  # type: ignore
         except Exception:  # pylint: disable=broad-except
             self.log.exception("Exception when executing SchedulerJob._run_scheduler_loop")
+            raise
```

Comment:
Noticed this was missing too. Without it, the SchedulerJob is always successful:

https://github.com/apache/airflow/blob/923bde2b917099135adfe470a5453f663131fd5f/airflow/jobs/base_job.py#L236-L249